execution_engine/src/core/engine_state/upgrade.rs

```diff
@@ -184,37 +185,49 @@ where
     /// Creates new system upgrader instance.
     pub(crate) fn new(
         new_protocol_version: ProtocolVersion,
+        old_protocol_version: ProtocolVersion,
         tracking_copy: Rc<RefCell<TrackingCopy<<S as StateProvider>::Reader>>>,
     ) -> Self {
         SystemUpgrader {
             new_protocol_version,
+            old_protocol_version,
             tracking_copy,
         }
     }
 
-    /// Bump major version for system contracts.
-    pub(crate) fn upgrade_system_contracts_major_version(
+    /// Check if the next version is a major version bump.
+    pub(crate) fn is_major_bump(&self) -> bool {
```

Comment:
Instead of reimplementing this method, let's use ones defined on `ProtocolVersion`:
```rust
old_protocol_version.check_next_version(&new_protocol_version).is_major_version()
```

Reply:
Using the method defined on `ProtocolVersion`